libs/ts-rest/core/src/lib/client.ts

```diff
@@ -327,10 +373,17 @@ export type InitClientReturn<
   TClientArgs extends ClientArgs
 > = RecursiveProxyObj<T, TClientArgs>;
 
+export type InitClientReturnNoUnknownStatus<
+  T extends AppRouter,
+  TClientArgs extends ClientArgs & { throwOnUnknownStatus: true }
+> = RecursiveProxyObjNoUnknownStatus<T, TClientArgs>;
+
 export const initClient = <T extends AppRouter, TClientArgs extends ClientArgs>(
```

Comment:
Regarding the other comment 

> This should be removed from this interface as it is being used by the rest of the libraries

```suggestion
export const initClient = <T extends AppRouter, TClientArgs extends ClientArgs & { throwOnUnknownStatus?: boolean; }>(
```
This should live here instead as it'll only be restricted to the vanilla client

Reply:
Done this. Pulled it out into a separate type so it could be reused in the two places needed.

Reply:
Perfect!